pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -511,7 +518,7 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			opconfig.DataPathMap{ContainerDataDir: dataPath},
 			c.spec.CephVersion.Image,
 			volumeMounts,
-			osdProps.resources,
+			c.osdResources(osdProps),
```

Comment:
is it possible to  change the `osdProps.resources` when its being initialised [here](https://github.com/rook/rook/blob/27b0295ac17b671d92cf9f57657bf72fb678557d/pkg/operator/ceph/cluster/osd/osd.go#L266)?  That might avoid calling `osdResources` multiple times in this file.

Reply:
right, let me try that.

Reply:
> is it possible to change the `osdProps.resources` when its being initialised [here](https://github.com/rook/rook/blob/27b0295ac17b671d92cf9f57657bf72fb678557d/pkg/operator/ceph/cluster/osd/osd.go#L266)? That might avoid calling `osdResources()` method multiple times in this file.

@sp98 thanks for suggestions it worked(I just needed to change below in that file [here](https://github.com/rook/rook/blob/27b0295ac17b671d92cf9f57657bf72fb678557d/pkg/operator/ceph/cluster/osd/osd.go#L727))